File: java/bowling-game-kata/test/BowlingGameKataTest.java
import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;

/**
 * Created by Peter Mösenthin.
 */
public class BowlingGameKataTest
{

	private BowlingGameKata bgk;

	@Before
	public void setUp() throws Exception {
		bgk = new BowlingGameKata();
	}
	
	private void rollMany(int n, int pins) {
		for (int i = 0; i < n; i++)
			bgk.roll(pins);
	}

	@Test
	public void testGutterGame() throws Exception {
		rollMany(20, 0);
		assertThat(bgk.score(), is(0));
	}

	@Test
	public void testAllOnes() throws Exception {
		rollMany(20,1);
		assertThat(bgk.score(), is(20));
	}

	@Test
	public void testOneSpare() throws Exception {
		rollSpare();
		bgk.roll(3);
		rollMany(17,0);
		assertThat(bgk.score(), is(16));
	}

	@Test
	public void testOneStrike() throws Exception {
		rollStrike();
		bgk.roll(3);
		bgk.roll(4);
		rollMany(16, 0);
		assertThat(bgk.score(), is(24));
	}

	@Test
	public void testPerfectGame() throws Exception {
		rollMany(12,10);
		assertThat(bgk.score(), is(300));
	}

	private void rollStrike() {
		bgk.roll(10);
	}

	private void rollSpare() {
		bgk.roll(5);
		bgk.roll(5);
	}
}

File: python/bowling-game-kata/bowling_game_kata_test.py
from bowling_game_kata import BowlingGameKata
import unittest

class BowlingGameKataTest(unittest.TestCase):

		def setUp(self):
			self.game = BowlingGameKata()

		def test_gutter_game(self):
			self.roll_many(20, 0)
			self.assertEquals(0, self.game.score())

		def test_all_ones(self):
			self.roll_many(20, 1)
			self.assertEquals(20, self.game.score())

		def test_one_spare(self):
			self.roll_spare()
			self.game.roll(3)
			self.roll_many(17, 0)
			self.assertEquals(16, self.game.score())

		def test_roll_strike(self):
			self.roll_strike()
			self.game.roll(3)
			self.game.roll(4)
			self.assertEquals(24, self.game.score())

		def test_perfect_game(self):
			self.roll_many(12, 10)
			self.assertEquals(300, self.game.score())

		def roll_strike(self):
			self.game.roll(10)

		def roll_spare(self):
			self.game.roll(5)
			self.game.roll(5)

		def roll_many(self, n, pins):
				for i in range(n):
					self.game.roll(pins)
